src/main/java/main/internet/shop/dao/jdbcimpl/ProductDaoJavaDataBaseConnectivityImpl.java

```diff
@@ -0,0 +1,113 @@
+            statement.setDouble(2, productPrice);
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't connect to MySQL", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * FROM products WHERE id = ?";
+        try (Connection connection = ConnectionUtils.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Product product = new Product(resultSet.getString("name"),
+                        resultSet.getDouble("price"));
+                product.setId(id);
+                return Optional.of(product);
+            } else {
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't connect to MySQL", e);
+        }
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products WHERE deleted = 0";
+        List<Product> allProducts = new ArrayList<>();
+        try (Connection connection = ConnectionUtils.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                allProducts.add(extractProductFromResultSet(resultSet));
+            }
+            return allProducts;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't connect to MySQL", e);
+        }
+    }
+
+    @Override
+    public Product update(Product item) {
+        String query = "UPDATE products SET name = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtils.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, item.getName());
+            statement.setDouble(2, item.getPrice());
+            statement.setLong(3, item.getId());
+            statement.executeUpdate();
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't connect to MySQL", e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "UPDATE products SET deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtils.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return true;
```

Comment:
should we always return true? let's check the result of statement.executeUpdate() and return the result of checking